utils/imagenet_topk_accuracy_driver.py

```diff
@@ -0,0 +1,332 @@
+
+## Opens and returns an image located at @param path using the PIL loader.
+def pil_loader(path):
+    # open path as file to avoid ResourceWarning
+    # (https://github.com/python-pillow/Pillow/issues/835)
+    with open(path, "rb") as img:
+        img = Image.open(img)
+        return img.convert("RGB")
+
+## Opens and returns an image located at @param path using the accimage loader.
+def accimage_loader(path):
+    import accimage
+    try:
+        return accimage.Image(path)
+    except IOError:
+        # Potentially a decoding problem, fall back to PIL.Image
+        return pil_loader(path)
+
+
+## Opens and returns an image located at @param path using either the accimage
+# loader or PIL loader.
+def default_image_loader(path):
+    if torchvision.get_image_backend() == "accimage":
+        return accimage_loader(path)
+    return pil_loader(path)
+
+def get_sorted_img_subdirs(validation_images_dir):
+    img_dir_paths = []
+    for img_dir in os.listdir(validation_images_dir):
+        dir_path = os.path.join(validation_images_dir, img_dir)
+        if os.path.isdir(dir_path):
+            img_dir_paths.append(img_dir)
+    img_dir_paths.sort()
+
+    return img_dir_paths
+
+
+## @returns two lists of the same length found in directory
+# @param validation_images_dir; the first list contains paths to all images
+# found, and the second list contains the corresponding labels of the image.
+def get_img_paths_and_labels(validation_images_dir):
+    img_subdirs = get_sorted_img_subdirs(validation_images_dir)
+
+    # Create lists holding paths to each image to be classified and the label
+    # for that image.
+    img_paths = []
+    img_labels = []
+    curr_label_idx = 0
+    for img_subdir in img_subdirs:
+        img_subdir_path = os.path.join(validation_images_dir, img_subdir)
+        for img in os.listdir(img_subdir_path):
+            full_img_path = os.path.join(img_subdir_path, img)
+            if os.path.isfile(full_img_path):
+                img_paths.append(full_img_path)
+                img_labels.append(curr_label_idx)
+        curr_label_idx = curr_label_idx + 1
+    return img_paths, img_labels
+
+## Given an image located at @param img_path, transform the image
+# and save it to the path @param path_to_new_img.
+def resize_and_save_image(img_path, path_to_new_img):
+    # Load the image.
+    img = default_image_loader(img_path)
+
+    # Use to Resize and CenterCrop the images to 224x224.
+    transform_resize = torchvision.transforms.Compose([
+        torchvision.transforms.Resize(256),
+        torchvision.transforms.CenterCrop(224),
+    ])
+
+    resized_img = transform_resize(img)
+    resized_img.save(path_to_new_img, format="png")
+
+## Used to pre-process an input set of images. Takes a string of a directory
+# @param validation_images_dir and saves the cropped subset of the images in a
+# subdirectory `processed/`, which must not yet exist.
+def save_centered_cropped_dataset(validation_images_dir):
```

Comment:
I think you can use this function for both "only-resize-and-save" and "resize-input-images" cases. Add a new argument "target_dir", and pass either `join(validation_images_dir, "processed")` or tmpdir (only run it if "resize-input-images"). 

So in case of "resize-input-images" you'll do all resizing in advance before starting classification. You'll reassign `img_paths` variable, and instead of calling `get_curr_img_paths` you'll just do: `img_paths[img_index, img_index + batch_size]`. Cleanup will happen automatically, when you remove tmpdir after `finally:`.

Reply:
>So in case of "resize-input-images" you'll do all resizing in advance before starting classification. 

Got it -- yeah that could work, but the downside here is the resized images must all exist at once. So for example the 50k images I was looking at would take up 4.2GB temporarily. I felt it was a waste to create that much temporary space on disk, but if you prefer that version in order to remove some duplicated code I can change it.

Reply:
Agree, that's a lot of space. But my understanding is that in current version we only clean up once, in the very end.

Reply:
Ah, actually the current version just reuses the same image name over and over again at [this line](https://github.com/pytorch/glow/pull/2009/files#diff-4ea927ddc0aaa413fc99b55398567f7aR182). So there are only ever `batch_size` temp images in the temp directory which is then cleaned up in the end. I can clarify that in the comments there if we proceed with this version.

Reply:
Oh, I totally missed this. Well, feel free to leave this logic then.